### PR TITLE
IReader+IWriter => Format

### PR DESCRIPTION
**Pull request by apavlenko**
as stated in #23

``` cpp
class VMF_EXPORT Format
{
public:

    typedef std::map<std::string, std::string> AttribMap;
    /*!
    * \brief Export various metadata stream items to a string representation. Used in particular for "stream.serialize()" implementation.
    */
    virtual std::string store(
        const MetadataSet& set,
        const std::vector<std::shared_ptr<MetadataSchema>>& schemas = {},
        const std::vector<std::shared_ptr<MetadataStream::VideoSegment>>& segments = {},
        const std::vector<Stat>& stats = {},
        const AttribMap& attribs = {} // nextId, checksum, etc
        ) = 0;

typedef union {
            struct { int metadata, schemas, segments, stats, attribs; };
            int cnt[5];
        } ParseCounters;
    /*!
    * \brief Deserialize input string to stream metadata and related stuff. Used in particular for "stream.deserialize(text)" implementation.
    */
    virtual ParseCounters parse(
        const std::string& text,
        std::vector<std::shared_ptr<MetadataInternal>>& metadata,
        std::vector<std::shared_ptr<MetadataSchema>>& schemas,
        std::vector<std::shared_ptr<MetadataStream::VideoSegment>>& segments,
        std::vector<Stat>& stats,
        AttribMap& attribs // nextId, checksum, etc
        ) = 0;
;
```

Simpler API.

### CODE REVIEW

**Comment by apavlenko** [comment]
@savuor , please add `FormatCompressed` implementation here

**Comment by apavlenko** [comment]
@anna-plyakina , please add Java wrapper and tests for `Format`

File: modules/vmfcore/include/vmf/format.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        const std::vector<std::shared_ptr<MetadataSchema>>& schemas = {},
+        const std::vector<std::shared_ptr<MetadataStream::VideoSegment>>& segments = {},
+        //const std::vector<Stat>& stats = {},
+        const AttribMap& attribs = AttribMap() // nextId, checksum, etc
+        ) = 0;
+
+    /*!
+    * \brief Deserialize input string to stream metadata and related stuff. Used in particular for "stream.deserialize(text)" implementation.
+    */
+    virtual std::array<int, 5> parse(
```
Comment (savuor) [inline modules/vmfcore/include/vmf/format.hpp]: Wouldn't it be better to change array<int, 5> to some other type, for example structure?
Referencing counters like `counter[3]` or `counter[0]` looks incomprehensible.

File: modules/vmfcore/java/src/com/intel/vmf/FormatJSON.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    protected FormatJSON (long addr)
+    {
+        if (addr == 0)
+            throw new java.lang.IllegalArgumentException("Native object address is NULL");
+        
+        nativeObj = addr;
+    }
+    
+	public FormatJSON() {
```
Comment (anna-khakimova) [inline modules/vmfcore/java/src/com/intel/vmf/FormatJSON.java]: Alignment is not so good)

File: modules/vmfcore/java/src/com/intel/vmf/FormatJSON.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		if(cntAtt>0) {
+			data.attrib = new HashMap<String, String>();
+			for(i=0; i<cntAtt; i++)
+				data.attrib.put(attribNames[i], attribVals[i]);
+		}
+		
+		return data;
+	}
+	
+    @Override
```
Comment (anna-khakimova) [inline modules/vmfcore/java/src/com/intel/vmf/FormatJSON.java:139]: Alignment is not so good.

File: modules/vmfcore/java/src/com/intel/vmf/FormatJSON.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    protected void finalize () throws Throwable 
+    {
+        if (nativeObj != 0)
+            n_delete (nativeObj);
+        
+        super.finalize();
+    }
+
+	private native static long n_FormatJSON();
```
Comment (anna-khakimova) [inline modules/vmfcore/java/src/com/intel/vmf/FormatJSON.java]: Alignment is not so good.

**Comment by savuor** [comment]
:+1:

**Pull request merged**